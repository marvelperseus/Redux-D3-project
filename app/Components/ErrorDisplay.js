import React, { Component } from 'react'
import PropTypes from 'prop-types'

/**
 * React component errors.
 */
class ErrorDisplay extends Component {
    
    render() {
        const { errorClass, error, errorInfo} = this.props
        return (
            <div className={errorClass}>
                <h2>Error</h2>
                <p className='color-red'>
                    {error && error.toString()}
                </p>
                <div>Component Stack Error Details:</div>
                <p className="small">
                {errorInfo && errorInfo.componentStack}
                </p>
            </div>
        )
    }
}

ErrorDisplay.propTypes = {
    errorClass : PropTypes.string,
    error      : PropTypes.object,
    errorInfo  : PropTypes.object
}

export default ErrorDisplay