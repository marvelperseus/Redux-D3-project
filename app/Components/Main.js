import React, { Component } from 'react'
import {
    Route,
    Switch,
    Redirect
} from 'react-router-dom'
import { Container } from 'semantic-ui-react'

import GraphHome from './Graphs/GraphHome'
import ErrorDisplay from './ErrorDisplay'

/**
 * Main Layout and Router
 */
class Main extends Component {

    constructor(props) {
        super(props)
        this.state = {
            error     : null,
            errorInfo : null
        }        
    }

     /**
     * Catches errors in child components
     * 
     * @param {*} error 
     * @param {*} info 
     */
    componentDidCatch(error, info) {
        // Display fallback UI
        this.setState({
            error     : error,
            errorInfo : info
        })
    }

    render() {
        return (
            <Container fluid>
                {this.state.error && <ErrorDisplay      
                    errorClass='graph-workspace'
                    error={this.state.error}
                    errorInfo={this.state.errorInfo}
                />}
                <Switch>
                    <Route exact path='/graphs' render={()=><GraphHome {...this.props}/>}/>
                    <Redirect to='/graphs' />
                </Switch>
            </Container>
        )
    }
}

export default Main
