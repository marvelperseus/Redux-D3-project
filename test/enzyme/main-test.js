import React from 'react'

import { shallow } from 'enzyme'

import Main from '../../app/Components/Main'
import SideNav from '../../app/Components/SideNav'
import Nav from '../../app/Components/Nav'
import ErrorDisplay from '../../app/Components/ErrorDisplay'

/**
 * Baseline non-Connected component tests using Enzyme.
 */
describe('The top level component render', () => {

    it('renders the <Main/> component', () => {
        const component = shallow(<Main/>)
        expect(component).toExist()
        expect(component.find('#hummingbird-app-main')).toHaveLength(1)
        expect(component.find(<SideNav/>).exists())
        expect(component.find(<Nav/>).exists())
        expect(component).toMatchSnapshot()
    })

    it('renders the <ErrorDisplay/> component', () => {
        const component = shallow(
        <ErrorDisplay 
            errorClass=''
            error={null}
            errorInfo={null}
            />)
        expect(component).toHaveLength(1)
    })


})

