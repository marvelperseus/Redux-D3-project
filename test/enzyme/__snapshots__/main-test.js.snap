// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The top level component render renders the <Main/> component 1`] = `
<div
  id="hummingbird-app-main"
>
  <Nav />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNav />
      <main
        className="col-md-10 ml-sm-auto col-lg-10 px-4"
        role="main"
      >
        <Switch>
          <Route
            exact={true}
            path="/graphs"
            render={[Function]}
          />
          <Redirect
            push={false}
            to="/graphs"
          />
        </Switch>
      </main>
    </div>
  </div>
</div>
`;
